folio-spring-i18n/src/main/java/org/folio/spring/service/TranslationService.java

```diff
@@ -0,0 +1,347 @@
+        return correspondingMap;
+      }
+    }
+
+    return this.getDefaultTranslation();
+  }
+
+  /**
+   * Wraps the {@link TranslationMap#format TranslationMap#format} method on the current translation.
+   * Equivalent to {@code getCurrentTranslation().format(...)}.
+   *
+   * <p>Format an ICU format string (found by its key), supplying a series of named arguments as key
+   * value pairs.  For example: {@code format("Hello {name}", "name", parameterValue)}</p>
+   *
+   * @param key the key of the format string.  You likely want to use a constant from
+   *   {@link TranslationKey TranslationKey} rather than hard-coding a string.
+   * @param args pairs of keys and values to interpolate
+   * @return the formatted string
+   */
+  public String format(String key, Object... args) {
+    return this.getCurrentTranslation().format(key, args);
+  }
+
+  /**
+   * Format a list of strings into one cohesive string.  For example, in English, {@code [A B C D]}
+   * would become {@code "A, B, C, and D"}.
+   *
+   * @param list a list of items to join
+   * @return the joined string
+   */
+  public String formatList(Collection<?> list) {
+    return ListFormatter.getInstance(getCurrentLocale()).format(list);
+  }
+
+  /**
+   * Get all available translations in the classpath.
+   *
+   * @return a list of available translation files
+   * @throws IllegalStateException if the translation files cannot be found
+   */
+  public List<TranslationFile> getAvailableTranslationFiles() {
+    try {
+      Map<String, List<Resource>> localeGroups = Arrays
+        .asList(
+          resourceResolver.getResources(
+            String.format(
+              TRANSLATIONS_CLASSPATH,
+              configuration.getTranslationDirectory()
+            )
+          )
+        )
+        .stream()
+        .filter(Resource::isReadable)
+        .collect(Collectors.groupingBy(Resource::getFilename));
+
+      List<TranslationFile> files = localeGroups
+        .entrySet()
+        .stream()
+        .map(entry -> new TranslationFile(entry.getValue()))
+        .toList();
+
+      log.info("Got translation files: " + files);
+
+      return files;
+    } catch (IOException e) {
+      log.error("Could not retrieve translation files:", e);
+      throw new IllegalStateException(
+        "Could not retrieve translation files",
+        e
+      );
+    }
+  }
+
+  /**
+   * Build the map of language -&gt; country -&gt; pattern resource based on the classpath contents.
+   *
+   * @return the map of languages to countries to pattern resources
+   */
+  public Map<String, Map<String, TranslationFile>> buildLanguageCountryPatternMap() {
+    log.info("Building translation file map");
+
+    Map<String, Map<String, TranslationFile>> map = new HashMap<>();
+
+    getAvailableTranslationFiles()
+      .forEach((TranslationFile file) -> {
+        LanguageRegionPair parts = file.getParts();
+        map
+          .computeIfAbsent(
```

Comment:
Tiniest of nits: I'd combine these 2 lines. We don't gain anything by splitting em up and it makes it very slightly harder to read